pkg/broker/metrics.go

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2019 The Knative Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package broker
+
+import (
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+)
+
+var (
+	// MeasureTriggerEventsTotal is a counter which records the number of events received
+	// by a Trigger.
+	MeasureTriggerEventsTotal = stats.Int64(
+		"knative.dev/eventing/trigger/measures/events_total",
+		"Total number of events received",
+		stats.UnitNone,
+	)
+
+	// MeasureTriggerFilterTime records the time spent filtering a message for a
+	// Trigger, in milliseconds.
+	MeasureTriggerFilterTime = stats.Int64(
```

Comment:
Does this include dispatching? Once we have CEL, we probably will want to be able to distinguish between time spent filtering and time spent dispatching.

Reply:
No this doesn't include dispatch time. It's only the time spend inside `shouldSendEvent`. We should probably add a separate dispatch time metric.

Reply:
I added `trigger_dispatch_time`. This includes both filtering and dispatch to subscriber, including subscriber latency.